### PR TITLE
Add proxy peering metrics to docs

### LINKED ISSUE #22998: Document proxy_peer metrics
`proxy_peer` prometheus metrics are not documented

Present here:

https://github.com/gravitational/teleport/blob/a5fa148070da1411df9586ffcb3dd5016694c863/lib/proxy/peer/servermetrics.go#L48

But not here:

https://goteleport.com/docs/reference/metrics/

### PR DESCRIPTION
**Pull request by rosstimothy**
Closes #22998

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

File: docs/pages/includes/metrics.mdx [hunk 1]
```diff
@@ -67,21 +67,34 @@
...
-| `teleport_reverse_tunnels_connected`      | gauge   | Teleport Proxy | Number of reverse SSH tunnels connected to the Teleport Proxy Service by Teleport instances.                                    |
-| `trusted_clusters`                        | gauge   | Teleport Proxy | Number of outbound connections to leaf clusters.                                                                                |
+| Name                                      | Type      | Component      | Description                                                                                                                     |
+| ----------------------------------------- | --------- | -------------- | ------------------------------------------------------------------------------------------------------------------------------- |
+| `failed_connect_to_node_attempts_total`   | counter   | Teleport Proxy | Number of failed SSH connection attempts to a Node. Use with `teleport_connect_to_node_attempts_total` to get the failure rate. |
```
Comment (ptgott) [inline docs/pages/includes/metrics.mdx:72]: Does `Node` refer to an SSH Service instance or any agent? We're phasing out the "Node" terminology, and the "Core Concepts" page doesn't mention this term.

File: docs/pages/includes/metrics.mdx [hunk 2]
```diff
@@ -67,21 +67,34 @@
...
+| `grpc_client_handled_total`               | counter   | Teleport Proxy | Total number of RPCs completed on the client, regardless of success or failure.                                                 |
+| `grpc_client_msg_received_total`          | counter   | Teleport Proxy | Total number of RPC stream messages received on the client.                                                                     |
+| `grpc_client_msg_sent_total`              | counter   | Teleport Proxy | Total number of gRPC stream messages sent by the client.                                                                        |
+| `proxy_connection_limit_exceeded_total`   | counter   | Teleport Proxy | Number of connections that exceeded the Proxy Service connection limit.                                                         |
+| `proxy_peer_client_dial_error_total`      | counter   | Teleport Proxy | Total number of errors encountered dialing peer proxies.                                                                        |
```
Comment (ptgott) [inline docs/pages/includes/metrics.mdx:79]:
```suggestion
| `proxy_peer_client_dial_error_total`      | counter   | Teleport Proxy | Total number of errors encountered dialing peer Proxy Service instances.                                                                        |
```

File: docs/pages/includes/metrics.mdx [hunk 3]
```diff
@@ -67,21 +67,34 @@
...
+| `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to proxy peer servers.                                                                    |
```
Comment (ptgott) [inline docs/pages/includes/metrics.mdx:80]:
```suggestion
| `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to peer Proxy Service instances.                                                                    |
```

File: docs/pages/includes/metrics.mdx [hunk 4]
```diff
@@ -67,21 +67,34 @@
...
+| `proxy_peer_client_rpc_total`             | counter   | Teleport Proxy | Total number of client RPC requests.                                                                                            |
+| `proxy_peer_client_rpc_duration_seconds`  | histogram | Teleport Proxy | Duration in seconds of RPCs sent by the client.                                                                                 |
+| `proxy_peer_client_message_sent_size`     | histogram | Teleport Proxy | Size of messages sent by the client.                                                                                            |
+| `proxy_peer_client_message_received_size` | histogram | Teleport Proxy | Size of messages received by the client.                                                                                        |
+| `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to proxy peer clients.                                                                    |
```
Comment (ptgott) [inline docs/pages/includes/metrics.mdx:86]:
```suggestion
| `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to Proxy Service peer clients.                                                                    |
```

File: docs/pages/includes/metrics.mdx [hunk 5]
```diff
@@ -67,21 +67,34 @@
...
+| `proxy_peer_server_rpc_duration_seconds`  | histogram | Teleport Proxy | Duration in seconds of RPCs sent by the server.                                                                                 |
+| `proxy_peer_server_message_sent_size`     | histogram | Teleport Proxy | Size of messages sent by the server.                                                                                            |
+| `proxy_peer_server_message_received_size` | histogram | Teleport Proxy | Size of messages received by the server.                                                                                        |
+| `proxy_ssh_sessions_total`                | gauge     | Teleport Proxy | Number of active sessions through this Proxy Service instance.                                                                  |
+| `proxy_missing_ssh_tunnels`               | gauge     | Teleport Proxy | Number of missing SSH tunnels. Used to debug if Nodes have discovered all Proxy Service instances.                              |
```
Comment (ptgott) [inline docs/pages/includes/metrics.mdx:93]: Same comment as above re: "Node"

File: docs/pages/includes/metrics.mdx [hunk 6]
```diff
@@ -67,21 +67,34 @@
...
+| `remote_clusters`                         | gauge     | Teleport Proxy | Number of inbound connections from leaf clusters.                                                                               |
+| `teleport_connect_to_node_attempts_total` | counter   | Teleport Proxy | Number of SSH connection attempts to a node. Use with `failed_connect_to_node_attempts_total` to get the failure rate.          |
```
Comment (ptgott) [inline docs/pages/includes/metrics.mdx:95]: Same comment as above re: "node"

File: docs/pages/includes/metrics.mdx [hunk 7]
```diff
@@ -67,27 +67,40 @@
...
+| `proxy_peer_client_rpc_total`             | counter   | Teleport Proxy | Total number of client RPC requests.                                                                                                     |
+| `proxy_peer_client_rpc_duration_seconds`  | histogram | Teleport Proxy | Duration in seconds of RPCs sent by the client.                                                                                          |
+| `proxy_peer_client_message_sent_size`     | histogram | Teleport Proxy | Size of messages sent by the client.                                                                                                     |
+| `proxy_peer_client_message_received_size` | histogram | Teleport Proxy | Size of messages received by the client.                                                                                                 |
+| `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to proxy peer clients.                                                                             |
```
Comment (ptgott) [inline docs/pages/includes/metrics.mdx:86]:
```suggestion
| `proxy_peer_server_connections`           | gauge     | Teleport Proxy | Number of currently opened connection to peer Proxy Service clients.                                                                             |
```

**Review by ptgott — APPROVED** [review]
Approved with suggestions for consistent naming

**Pull request merged**